### PR TITLE
Dev UI - list reactive messaging channels

### PR DESCRIPTION
**Pull request by mkouba**
- also replace recorder methods with a synthetic bean

### CODE REVIEW

**Comment by mkouba** [comment]
Screenshot from 2021-01-25 13-35-44

**Comment by cescoffier** [comment]
I wonder if it would not be simpler to retrieve the graph from reactive messaging from @Inject Wiring wiring.

**Comment by cescoffier** [comment]
Here is the bean that could be interesting: https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/wiring/Wiring.java

The getGraph method return the graph.

**Comment by mkouba** [comment]
> I wonder if it would not be simpler to retrieve the graph from reactive messaging from @Inject Wiring wiring.

@cescoffier That was my first attempt ;-). Unfortunately, `Wiring.Component#toString()` does not always contain the info I need, not to mention that subinterfaces are package-private and thus not usable.

I can send a PR to SM Reactive Messaging to change the visibility of subinterfaces but even then for example `EmitterConfiguration` does not contain the injection point target (field/method) etc.

**Comment by cescoffier** [comment]
Ah, yes. We can definitely improve that.

**Comment by mkouba** [comment]
> Ah, yes. We can definitely improve that.

Ok, I'll send a PR tomorrow and I'll create a quarkus issue to update the Dev UI components once the new SM RM is released...

**Comment by cescoffier** [comment]
Great! Let's talk tomorrow to see how we can improve the wiring API to get what you need from it.

**Review by cescoffier — APPROVED** [review]

File: extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/devconsole/DevConsoleProcessor.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public DevConsoleRuntimeTemplateInfoBuildItem collectInfos() {
+        return new DevConsoleRuntimeTemplateInfoBuildItem("reactiveMessagingInfos",
+                new DevReactiveMessagingInfosSupplier());
+    }
+
+    @Record(STATIC_INIT)
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public void collectInjectionInfo(DevConsoleRecorder recorder, BeanDiscoveryFinishedBuildItem beanDiscoveryFinished) {
+        Map<String, String> emitters = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (mkouba) [reply extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/devconsole/DevConsoleProcessor.java:33]: Yes, but no injection point target info...

File: extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/devconsole/Connectors.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    public Connectors() {
+        incomingConnectors = new HashMap<>();
+        outgoingConnectors = new HashMap<>();
+    }
+
+    void collect(@Observes StartupEvent event, @Any Instance<IncomingConnectorFactory> incoming,
+            @Any Instance<OutgoingConnectorFactory> outgoing, Config config) {
+
+        Set<String> connectors = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (mkouba) [reply extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/devconsole/Connectors.java:41]: Yes, but no way to filter connectors and extract the name so far. We'll improve SM RM API ;-).

**Pull request merged**